src/api/api.js

```diff
@@ -19,441 +53,322 @@ export const getMessageConversations = ({
-        .catch(error => {
-            throw error
-        })
-
-/* Recipient search */
-const MAX_RECIPIENT = 10
-const searchOrganisationUnits = searchValue =>
-    getD2Instance()
-        .then(instance =>
-            instance.Api.getApi().get('organisationUnits', {
-                fields: 'id, displayName',
-                pageSize: MAX_RECIPIENT,
-                filter: [`displayName:token:${searchValue}`, 'users:gte:1'],
-            })
-        )
-        .then(result => result)
-        .catch(error => {
-            throw error
-        })
-
-const searchUserGroups = searchValue =>
-    getD2Instance()
-        .then(instance =>
-            instance.Api.getApi().get('userGroups', {
-                fields: 'id, displayName',
-                pageSize: MAX_RECIPIENT,
-                filter: [`displayName:token:${searchValue}`],
-            })
-        )
-        .then(result => result)
-        .catch(error => {
-            throw error
-        })
-
-export const searchRecipients = ({
+    await engine.mutate({
+        resource: 'messageConversations',
+        type: 'create',
+        data: {
+            subject,
+            users,
+            userGroups,
+            organisationUnits,
+            text,
+            attachments,
+        },
+    })
+
+export const sendFeedbackMessage = async (subject, text) =>
+    await engine.mutate({
+        resource: 'messageConversations/feedback',
+        type: 'create',
+        params: { subject },
+        data: text,
+    })
+
+export const replyMessage = async ({
+    message,
+    internalReply,
+    attachments,
+    id,
+}) => {
+    const params = { internal: internalReply }
+
+    if (attachments.length > 0) {
+        params.attachments = attachments
+    }
+
+    return await engine.mutate({
+        resource: `messageConversations/${id}`,
+        type: 'create',
+        params,
+        data: message,
+    })
+}
+
+export const deleteMessageConversation = async (
+    messageConversationId,
+    currentUserId
+) =>
+    await engine.mutate({
+        resource: `messageConversations/${messageConversationId}/${currentUserId}`,
+        type: 'delete',
+    })
+
+export const markRead = async markedReadConversations =>
+    await engine.mutate({
+        resource: 'messageConversations/read',
+        type: 'create',
+        data: markedReadConversations,
+    })
+
+export const markUnread = async markedUnreadConversations =>
+    await engine.mutate({
+        resource: 'messageConversations/unread',
+        type: 'create',
+        data: markedUnreadConversations,
+    })
+
+export const isInFeedbackRecipientGroup = async currentUser => {
+    const {
+        configuration: { feedbackRecipients },
+    } = await engine.query({
+        configuration: {
+            resource: 'configuration',
+        },
+    })
+    const authorized =
+        currentUser.authorities.includes('ALL') ||
+        currentUser.userGroups.some(({ id }) => id === feedbackRecipients.id)
```

Comment:
This seems like a good candidate for a selector.

Reply:
IMO that's out-of-scope for this PR, I want to stick to replacing d2 by the data engine.

Reply:
Ok. I don't mean an elaborate setup or anything. It's just that I tend to prefer selectors for any kind of derived state for readability. But I'll leave it you.